### ISSUE TITLE
Log execution time of all CRUD APIs

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#214]
## Problem

Users of this library have no easy way of knowing how long each query `find()` or insert/update/delete `commit()` functions take to execute, which would be very useful to know when dealing with very large data sets (e.g. 100,000 contacts).

## Solution

Log the execution time in the `LoggerRegistry`, which currently looks like;

```kotlin
class LoggerRegistry(logger: Logger) {
    ...
        override fun onPreExecute(api: CrudApi) {
            logRedactable(api)
        }

        override fun onPostExecute(result: CrudApi.Result) {
            logRedactable(result)
        }

        private fun logRedactable(redactable: Redactable) {
            logger.log(redactable.redactedCopyOrThis(logger.redactMessages).toString())
        }
    ...
}
```

I'm thinking of adding `private val apiExecutionStartTimeMillis: MutableMap<CrudApi, Long>`. We can do something like...

```kotlin
override fun onPreExecute(api: CrudApi) {
    apiExecutionStartTimeMillis[api] = System.currentTimeMillis()
    logger.log(api.redactedCopyOrThis(logger.redactMessages).toString())
}

override fun onPostExecute(api: CrudApi, result: CrudApi.Result) {
    val executionTime = apiExecutionStartTimeMillis.remove(api)

    logger.log(
        """
           ${api.redactedCopyOrThis(logger.redactMessages)}
           execution time: $executionTime
        """.trimIndent()
        )
}
```

This will require refactoring ALL calls to `onPostExecute` to add the `api: CrudApi` as a parameter. It's a lot of redundant work but having the `api` and the `result` available in the same function will be very useful for a lot of other things and should have been the initial implementation anyways 😅

**Comment by vestrel00** [comment]
> Gotta get that timer in there!

Hey! Been long time my dude! Thanks for helping me out with my passion project!

**Comment by vestrel00** [comment]
This is included in [release 0.2.1](https://github.com/vestrel00/contacts-android/releases/tag/0.2.1)!
